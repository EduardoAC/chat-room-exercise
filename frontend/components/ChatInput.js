import React, { Component } from 'react';
import PropTypes from 'prop-types';

import '../styles/ChatInput.css';

class ChatInput extends Component {
  constructor(props) {
    super(props);
    this.state = { message: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(event) {
    const { socket } = this.props;
    event.preventDefault();
    socket.emit('message',  this.state.message);
  }

  handleChange(event) {
    this.setState({message: event.target.value});
  }
  render() {
    const { message } = this.state;
    return (
      <form className="chat-message-form" onSubmit={this.handleSubmit}>
        <input
          className="chat-message-form__message"
          autoComplete="off"
          type="text"
          value={message}
          onChange={this.handleChange}
        />
        <input className="chat-message-form__submit" type="submit" value="Send" />
      </form>
    );
  }
}

ChatInput.propTypes = {
  socket: PropTypes.shape({}).isRequired,
}

export default ChatInput;
