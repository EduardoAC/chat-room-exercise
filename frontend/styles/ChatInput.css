.chat-message-form {
  background: #bfbfbf;
  border-radius: 0 0 4px 4px;
  padding: 9px;
  width: 100%;
}

.chat-message-form__message {
  border: none;
  padding: 10px;
  width: 225px;
}

.chat-message-form__submit {
  background: #76cdd3;
  border: none;
  color: #FFF;
  font-weight: bold;
  padding: 10px;
  width: 55px;
}
