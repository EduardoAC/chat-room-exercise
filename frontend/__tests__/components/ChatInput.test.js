import React from 'react';
import { shallow, mount } from 'enzyme';

import ChatInput from '../../components/ChatInput';

describe('<ChatInput />', () => {
  it('Component renders correctly', () => {
    const chatInputWrapper = shallow(<ChatInput socket={{}} />);
    expect(chatInputWrapper).toMatchSnapshot();
  });

  it('stores the message typed', () => {
    const chatInputWrapper = mount(<ChatInput socket={{ emit: jest.fn() }} />);
    const messageInputBox = chatInputWrapper.find('.chat-message-form__message');
    messageInputBox.simulate('change', { target: { value: 'HELLO' } });
    expect(chatInputWrapper.state().message).toBe('HELLO');
  });

  it('submits the right message', () => {
    const emitFn = jest.fn();
    const chatInputWrapper = mount(<ChatInput socket={{ emit: emitFn }} />);
    const messageInputBox = chatInputWrapper.find('.chat-message-form__message');
    messageInputBox.simulate('change', { target: { value: 'HELLO' } });
    const submitButton = chatInputWrapper.find('.chat-message-form__submit');
    submitButton.simulate('submit');
    expect(emitFn).toHaveBeenCalledTimes(1);
    expect(emitFn).toHaveBeenCalledWith('message', 'HELLO');
  })
});
