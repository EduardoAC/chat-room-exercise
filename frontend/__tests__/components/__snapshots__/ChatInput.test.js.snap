// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatInput /> Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    className="chat-message-form"
    onSubmit={[Function]}
>
    <input
        autoComplete="off"
        className="chat-message-form__message"
        onChange={[Function]}
        type="text"
        value=""
    />
    <input
        className="chat-message-form__submit"
        type="submit"
        value="Send"
    />
</form>,
  "nodes": Array [
    <form
      className="chat-message-form"
      onSubmit={[Function]}
>
      <input
            autoComplete="off"
            className="chat-message-form__message"
            onChange={[Function]}
            type="text"
            value=""
      />
      <input
            className="chat-message-form__submit"
            type="submit"
            value="Send"
      />
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ChatInput
        socket={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ChatInput {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "handleSubmit": [Function],
        "props": Object {
          "socket": Object {},
        },
        "refs": Object {},
        "state": Object {
          "message": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          className="chat-message-form"
          onSubmit={[Function]}
>
          <input
                    autoComplete="off"
                    className="chat-message-form__message"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
          <input
                    className="chat-message-form__submit"
                    type="submit"
                    value="Send"
          />
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          className="chat-message-form"
          onSubmit={[Function]}
>
          <input
                    autoComplete="off"
                    className="chat-message-form__message"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
          <input
                    className="chat-message-form__submit"
                    type="submit"
                    value="Send"
          />
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ChatInput
    socket={Object {}}
/>,
}
`;
